listener.go

```diff
@@ -258,23 +269,46 @@ func (l *MetaTxnListener) WaitForMetaTxn(ctx context.Context, metaTxnID MetaTxnI
 		Str("meta-tx", string(metaTxnID)).
 		Msgf("Receipt not found among past receipts. Now listening..")
 
-	// Wait for receipt or
-	// context deadline
-	for {
-		select {
-		case <-ctx.Done():
-			err := ctx.Err()
-			if errors.Is(err, context.DeadlineExceeded) {
-				return 0, nil, fmt.Errorf("waiting for meta transaction timeout for %v: %w", metaTxnID, err)
-			} else if err != nil {
-				return 0, nil, fmt.Errorf("failed waiting for meta transaction for %v: %w", metaTxnID, err)
-			} else {
-				return 0, nil, nil
-			}
-		case receipt := <-sub.ch:
-			if receipt.MetaTxnID == metaTxnID {
-				return receipt.Status, receipt.TxnReceipt, nil
+	// Listen for new receipts
+	sub := l.subscribe()
```

Comment:
note, I've set a read lock on the pastReceipts and moved subscriber below here. We need the read lock on past receipts anyways and also the subscriber list, so may as well take advantage of it here slightly.